### ISSUE TITLE
Error: id already used when including two bases that rename same resource in parent

### ISSUE THREAD

**Issue by jessesuen** [kubernetes-sigs/kustomize#1083]
I have the following kustomize structure:

```
.
├── aggregate                // has guestbook-a and guestbook-b as bases
│   └── kustomization.yaml
├── guestbook-a              // has template as a base, renamed using json6902
│   ├── kustomization.yaml
│   └── rename.yaml
├── guestbook-b              // has template as a base, renamed using json6902
│   ├── kustomization.yaml
│   └── rename.yaml
└── template                 // single resource
    ├── guestbook.yaml
    └── kustomization.yaml
```

When trying to build the aggregate directory (which simply includes guestbook-a and guestbook-b as bases), it fails with:

```bash
$ kustomize build aggregate
Error: ../guestbook-b: id '"~G_v1_Pod|~X|~P|guestbook|~S"' already used
```

* aggregate/kustomization.yaml
```yaml
bases:
- ../guestbook-a
- ../guestbook-b
```

Both of the guestbook-a and guestbook-b directories rename the same resource in the template base using json6902:
* guestbook-a/rename.yaml
```yaml
- op: replace
  path: /metadata/name
  value: guestbook-a
```

* guestbook-b/rename.yaml
```yaml
- op: replace
  path: /metadata/name
  value: guestbook-b
```

The failing example can be reproduced here:
https://github.com/jessesuen/k8s-deployments/tree/master/kustomize-json6902-rename-already-used

**Comment by jessesuen** [comment]
Possibly related: https://github.com/kubernetes-sigs/kustomize/pull/971

**Comment by Liujingfang1** [comment]
This has been fixed by #1154. Please check out v3.0.0-pre.
